RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -0,0 +1,419 @@
+              << " etawidth "                 << vars[4] << std::endl
+              << " phiwidth "                 << vars[5] << std::endl
+              << " hoe "                      << vars[6] << std::endl
+              << " kfhits "                   << vars[7] << std::endl
+              << " kfchi2 "                   << vars[8] << std::endl
+              << " gsfchi2 "                  << vars[9] << std::endl
+              << " fbrem "                    << vars[10] << std::endl
+              << " gsfhits "                  << vars[11] << std::endl
+              << " expectedMissingInnerHits " << vars[12] << std::endl
+              << " convVtxFitProbability "    << vars[13] << std::endl
+              << " eop "                      << vars[14] << std::endl
+              << " eleeopout "                << vars[15] << std::endl
+              << " oneOverEminusOneOverP "                  << vars[16] << std::endl
+              << " deta "                     << vars[17] << std::endl
+              << " dphi "                     << vars[18] << std::endl
+              << " detacalo "                 << vars[19] << std::endl;
+    if (withIso_) {
+      std::cout << " ele_pfPhotonIso "          << vars[20] << std::endl
+                << " ele_pfChargedHadIso "      << vars[21] << std::endl
+                << " ele_pfNeutralHadIso "      << vars[22] << std::endl
+                << " rho "                      << vars[23] << std::endl
+                << " preShowerOverRaw "         << vars[24] << std::endl;
+    }
+    else {
+      std::cout << " rho "                      << vars[20] << std::endl
+                << " preShowerOverRaw "         << vars[21] << std::endl;
+    }
+    std::cout << " ### MVA " << result << std::endl << std::endl;
+  }
+
+  return result;
+}
+
+int ElectronMVAEstimatorRun2Fall17::findCategory( const edm::Ptr<reco::Candidate>& particle) const {
+
+  // Try to cast the particle into a reco particle.
+  // This should work for both reco and pat.
+  const edm::Ptr<reco::GsfElectron> eleRecoPtr = ( edm::Ptr<reco::GsfElectron> )particle;
+  if( eleRecoPtr.get() == nullptr ) {
+    throw cms::Exception("MVA failure: ")
+      << " given particle is expected to be reco::GsfElectron or pat::Electron," << std::endl
+      << " but appears to be neither" << std::endl;
+  }
+  return findCategory(eleRecoPtr.get());
+}
+
+int ElectronMVAEstimatorRun2Fall17::findCategory( const reco::GsfElectron * eleRecoPtr ) const {
+  float pt = eleRecoPtr->pt();
+  float eta = eleRecoPtr->superCluster()->eta();
+
+  //
+  // Determine the category
+  //
+  int  iCategory = UNDEFINED;
+  const float ptSplit = 10;   // we have above and below 10 GeV categories
+  const float ebSplit = 0.800;// barrel is split into two regions
+  const float ebeeSplit = 1.479; // division between barrel and endcap
+
+  if (pt < ptSplit && std::abs(eta) < ebSplit) {
+    iCategory = CAT_EB1_PT5to10;
```

Comment:
I would rather call ```CAT_EB1_PTLow``` (or similar), and ```CAT_EB1_PTHig``` below, in order not to have carved in the name of the variable the pt thresholds

Reply:
Done.